### PR TITLE
media-libs/libsdl2: wire up tests

### PR DESCRIPTION
**Pull request by matoro**

### CODE REVIEW

File: media-libs/libsdl2/libsdl2-2.28.5.ebuild [hunk 1]
```diff
@@ -110,6 +111,7 @@ src_prepare() {
 	# Unbundle some headers.
 	rm -r src/video/khronos || die
 	ln -s "${ESYSROOT}/usr/include" src/video/khronos || die
+	use vulkan || sed -i '/testvulkan$(EXE) \\/d' "test/Makefile.in" || die
```
Comment (mgorny) [inline media-libs/libsdl2/libsdl2-2.28.5.ebuild:114]: Please don't do these `||` chains, they are horribly unreadable and prone to mistakes in the future.

File: media-libs/libsdl2/libsdl2-2.28.5.ebuild [hunk 2]
```diff
@@ -205,10 +207,27 @@ multilib_src_configure() {
...
+		SDL_CFLAGS="-I${S}/include"
+		SDL_LIBS="-L${BUILD_DIR}/build/.libs -lSDL2"
+		ac_cv_lib_SDL2_ttf_TTF_Init=no
+		CFLAGS="${CPPFLAGS} ${CFLAGS} ${LDFLAGS}"
+	)
+
+	use test && \
+		mkdir "${BUILD_DIR}/test" && \
+		cd "${BUILD_DIR}/test" && \
+		ECONF_SOURCE="${S}/test" econf "${mytestargs[@]}"
```
Comment (mgorny) [inline media-libs/libsdl2/libsdl2-2.28.5.ebuild:225]: This really asks for an `if`, also it doesn't handle errors correctly.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by orbea** [comment]
This doesn't build on my system.
```
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: /var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/build/.libs/libSDL2_test.a(SDL_test_memory.o): in function `SDL_TrackAllocation':
SDL_test_memory.c:(.text+0x210): undefined reference to `unw_getcontext'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x21b): undefined reference to `unw_init_local'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x25c): undefined reference to `unw_step'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x270): undefined reference to `unw_get_reg'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x295): undefined reference to `unw_get_proc_name'
collect2: error: ld returned 1 exit status
make: *** [Makefile:365: testcustomcursor] Error 1
make: *** Waiting for unfinished jobs....
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: /var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/build/.libs/libSDL2_test.a(SDL_test_memory.o): in function `SDL_TrackAllocation':
SDL_test_memory.c:(.text+0x210): undefined reference to `unw_getcontext'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x21b): undefined reference to `unw_init_local'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x25c): undefined reference to `unw_step'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x270): undefined reference to `unw_get_reg'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x295): undefined reference to `unw_get_proc_name'
collect2: error: ld returned 1 exit status
make: *** [Makefile:182: testdraw2] Error 1
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: /var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/build/.libs/libSDL2_test.a(SDL_test_memory.o): in function `SDL_TrackAllocation':
SDL_test_memory.c:(.text+0x210): undefined reference to `unw_getcontext'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x21b): undefined reference to `unw_init_local'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x25c): undefined reference to `unw_step'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x270): undefined reference to `unw_get_reg'
/usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x295): undefined reference to `unw_get_proc_name'
collect2: error: ld returned 1 exit status
make: *** [Makefile:158: testautomation] Error 1
make: Leaving directory '/var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/test'
```
[build.log](https://github.com/gentoo/gentoo/files/13403588/build.log)

**Comment by matoro** [comment]
> This doesn't build on my system.
> 
> ```
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: /var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/build/.libs/libSDL2_test.a(SDL_test_memory.o): in function `SDL_TrackAllocation':
> SDL_test_memory.c:(.text+0x210): undefined reference to `unw_getcontext'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x21b): undefined reference to `unw_init_local'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x25c): undefined reference to `unw_step'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x270): undefined reference to `unw_get_reg'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x295): undefined reference to `unw_get_proc_name'
> collect2: error: ld returned 1 exit status
> make: *** [Makefile:365: testcustomcursor] Error 1
> make: *** Waiting for unfinished jobs....
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: /var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/build/.libs/libSDL2_test.a(SDL_test_memory.o): in function `SDL_TrackAllocation':
> SDL_test_memory.c:(.text+0x210): undefined reference to `unw_getcontext'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x21b): undefined reference to `unw_init_local'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x25c): undefined reference to `unw_step'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x270): undefined reference to `unw_get_reg'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x295): undefined reference to `unw_get_proc_name'
> collect2: error: ld returned 1 exit status
> make: *** [Makefile:182: testdraw2] Error 1
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: /var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/build/.libs/libSDL2_test.a(SDL_test_memory.o): in function `SDL_TrackAllocation':
> SDL_test_memory.c:(.text+0x210): undefined reference to `unw_getcontext'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x21b): undefined reference to `unw_init_local'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x25c): undefined reference to `unw_step'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x270): undefined reference to `unw_get_reg'
> /usr/lib/gcc/x86_64-gentoo-linux-musl/13/../../../../x86_64-gentoo-linux-musl/bin/ld: SDL_test_memory.c:(.text+0x295): undefined reference to `unw_get_proc_name'
> collect2: error: ld returned 1 exit status
> make: *** [Makefile:158: testautomation] Error 1
> make: Leaving directory '/var/tmp/portage/media-libs/libsdl2-2.28.5/work/SDL2-2.28.5-abi_x86_64.amd64/test'
> ```
> 
> [build.log](https://github.com/gentoo/gentoo/files/13403588/build.log)

Got it, this is due to sys-libs/llvm-libunwind, which is what [this Debian workaround is for](https://salsa.debian.org/sdl-team/libsdl2/-/blob/debian/latest/debian/rules#L78-79).  Pushed a fix to include this, see if that works.

File: media-libs/libsdl2/libsdl2-2.28.5.ebuild (hunk 1 above)
Comment (matoro) [reply media-libs/libsdl2/libsdl2-2.28.5.ebuild]: Done

File: media-libs/libsdl2/libsdl2-2.28.5.ebuild (hunk 2 above)
Comment (matoro) [reply media-libs/libsdl2/libsdl2-2.28.5.ebuild]: Done

**Comment by orbea** [comment]
@matoro That worked for me, thanks!

**Pull request closed without merging**